src/generation.rs

```diff
@@ -846,17 +876,59 @@ impl GenerationScope {
                 .cloned()
                 .collect::<Vec<_>>();
             for scope in scope_names {
-                self.wasm_lib()
-                    .raw(&format!("pub mod {scope};"))
-                    .push_import(&scope, "*", None);
+                self.wasm_lib().raw(&format!("pub mod {scope};"));
             }
             let mut merged_wasm_scope = codegen::Scope::new();
             merged_wasm_scope.append(self.wasm_lib());
             for (scope, content) in self.wasm_scopes.iter_mut() {
+                // might not exist if we don't use stuff from other scopes
+                if let Some(scope_imports) = rust_imports.get(scope) {
+                    for (import_scope, idents) in scope_imports.iter() {
+                        let import_scope = if import_scope == "lib" {
+                            Cow::from("super")
+                        } else if scope == "lib" {
+                            Cow::from(import_scope)
+                        } else {
+                            Cow::from(format!("crate::{import_scope}"))
+                        };
+                        #[allow(clippy::comparison_chain)]
+                        if idents.len() > 1 {
+                            content.push_import(
+                                import_scope,
+                                format!(
+                                    "{{{}}}",
+                                    idents
+                                        .iter()
+                                        .map(|i| i.to_string())
+                                        .collect::<Vec<_>>()
+                                        .join(", ")
+                                ),
+                                None,
+                            );
+                        } else if idents.len() == 1 {
+                            content.push_import(
+                                import_scope,
+                                idents.first().unwrap().to_string(),
+                                None,
+                            );
+                        }
+                    }
+                }
```

Comment:
Seems like a lot of code duplication

Reply:
Will fix in next PR as this PR already contains a lot of different stuff